File: backend/app/models/enums/enums.py
from enum import StrEnum


class UserRole(StrEnum):
    """Represents the role of the user"""

    ADMIN = "ADMIN"
    CLIENT = "CLIENT"
    TECHNICIAN = "TECHNICIAN"


class AdminRole(StrEnum):
    """Represents the role of the admin"""

    SUPER_ADMIN = "SUPER_ADMIN"
    SUPPORT_ADMIN = "SUPPORT_ADMIN"
    CONTENT_ADMIN = "CONTENT_ADMIN"


class BookingStatus(StrEnum):
    """Represents the status of a booking"""

    REQUESTED = "REQUESTED"
    ACCEPTED = "ACCEPTED"
    REJECTED = "REJECTED"
    IN_PROGRESS = "IN_PROGRESS"
    COMPLETED = "COMPLETED"
    CANCELLED = "CANCELLED"

File: frontend/src/lib/types/enums.ts
import { z } from "zod";

export const UserRoleSchema = z.enum(["ADMIN", "CLIENT", "TECHNICIAN"]);

export const AdminRoleSchema = z.enum(["SUPER_ADMIN", "SUPPORT_ADMIN", "CONTENT_ADMIN"]);

export const BookingStatusSchema = z.enum([
    "REQUESTED",
    "ACCEPTED",
    "REJECTED",
    "IN_PROGRESS",
    "CANCELLED",
    "COMPLETED",
])

export const PaymentStatusSchema = z.enum(["PENDING", "EZCROW", "COMPLETED", "RETURNED"]);

export type UserRole = z.infer<typeof UserRoleSchema>;
export type AdminRole = z.infer<typeof AdminRoleSchema>;
export type BookingStatus = z.infer<typeof BookingStatusSchema>;
export type PaymentStatus = z.infer<typeof PaymentStatusSchema>;
